libsolidity/formal/SMTEncoder.cpp

```diff
@@ -780,6 +783,32 @@ void SMTEncoder::visitRequire(FunctionCall const& _funCall)
 	addPathImpliedExpression(expr(*args.front()));
 }
 
+void SMTEncoder::visitBytesConcat(FunctionCall const& _funCall)
+{
+	auto const& args = _funCall.sortedArguments();
+	auto const& [name, inTypes, outType] = state().bytesConcatFunctionTypes(&_funCall);
+	// bytes.concat call with no arguments returns an empty array
+	if (args.size() == 0)
+	{
+		defineExpr(_funCall, smt::zeroValue(TypeProvider::bytesMemory()));
+		return;
+	}
+
+	// bytes.concat with single argument of type bytes memory
+	if (args.size() == 1 && args.front()->annotation().type->category() == frontend::Type::Category::Array)
+	{
+		defineExpr(_funCall, expr(*args.front(), TypeProvider::bytesMemory()));
+		return;
```

Comment:
Why do we need to separate the case with 0 and 1 arguments from >=2?

Reply:
Because it would best fit solidity semantics if `bytes.concat` is defined as an expression of value zerobytes for empty argument list, and the argument itself if there is exactly 1 argument.